src/Polly.Core/Timeout/TimeoutResilienceStrategyBuilderExtensions.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using Polly.Builder;
+
+namespace Polly.Timeout;
+
+/// <summary>
+/// Extension methods for adding timeouts to <see cref="ResilienceStrategyBuilder"/>.
+/// </summary>
+public static class TimeoutResilienceStrategyBuilderExtensions
+{
+    /// <summary>
+    /// Adds a timeout resilience strategy to the builder.
+    /// </summary>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="timeout">The timeout value. This value should be greater than <see cref="TimeSpan.Zero"/> and less than <see cref="System.Threading.Timeout.InfiniteTimeSpan"/>.</param>
```

Comment:
Should we validate this, rather than just give guidance?

Reply:
Yup definitely, if we are OK with the API shape I'll cover all edge-cases.

Reply:
Done.